src/types/validator.ts

```diff
@@ -0,0 +1,42 @@
+import {bytes48, Shard, Slot, uint64} from "./primitive";
+import {SimpleContainerType} from "@chainsafe/ssz";
+
+export interface ValidatorDuty {
```

Comment:
I think we shouldn't couple the committee assignment to the block production assignment.

It seems like the committee only needs to be checked once per epoch (and can be looked up one epoch beforehand), whereas the block production assignment must be checked every slot.

Reply:
you should comment on the issue with this! very valuable

Reply:
Chatter from gitter:

Paul Hauner @paulhauner 18:34
@GregTheGreek we weren’t quite sure about the best choice between IndexedAttestation and AttestationData. If you send AttestationData the validator client also needs to know the committee size to build the bitfield. We figured sending IndexedAttestation sends redundant data (an empty sig), however it moves one more concern away from the validator (how long should the bitfield be). Keen to hear thoughts.

Danny Ryan @djrtwo 18:38
Why not just send AttestationData and have the node package it up properly?
I suppose the validator would have to send back AttestationData and a different sig struct
but would sufficiently mask all the container/bitfield complexity

Paul Hauner @paulhauner 18:45
That could also work. You’d also have to send back your committee index and custody bit.

Danny Ryan @djrtwo 19:01
yes on the custody bit. you could send your validator index rather than committee index and the node could calculate the committee index. You wouldn’t necessarily know the committee index
seems like IndexedAttestation might be the appropriate container for this communication
(as you previously noted :)
It could also be node sends AttestationData, validator sends back IndexedAttestation, but the assymetry there isn’t great